nxcomp/configure.ac

```diff
@@ -0,0 +1,97 @@
+# ***************************************************************************
+# *** configure.ac for nxcomp                                         ***
+# ***************************************************************************
+
+m4_define([nxcomp_version], m4_esyscmd([tr -d '\n' < VERSION]))
+
+# Initialize Autoconf
+AC_PREREQ(2.60)
+
+AC_INIT([libXcomp], [nxcomp_version], [https://github.com/ArcticaProject/nx-libs/issues])
+AC_CONFIG_AUX_DIR([build-aux])
+AC_PROG_CXX
+AC_CONFIG_SRCDIR([Makefile.am])
+AC_CONFIG_HEADERS([config.h])
+AC_CONFIG_MACRO_DIR([m4])
+
+AM_INIT_AUTOMAKE([foreign no-dist-gzip dist-xz])
+
+# Initialize libtool
+AC_PROG_LIBTOOL
+
+COMP_VERSION=nxcomp_version
+AC_SUBST([COMP_VERSION])
+
+LT_COMP_VERSION=[`echo $COMP_VERSION | sed -r -e 's/^([0-9]+\.[0-9]+\.[0-9]+).*$/\1/' -e 's/\./:/g'`] 
+AC_SUBST([LT_COMP_VERSION])
+
+PKG_CHECK_MODULES(JPEG, libjpeg)
+PKG_CHECK_MODULES(PNG, libpng)
+PKG_CHECK_MODULES(Z, zlib)
+
+AC_LANG([C++])
+NX_COMPILER_BRAND
+NX_COMPILER_FLAGS
+
+NX_BUILD_ON_CYGWIN32
+NX_BUILD_ON_AMD64
+NX_BUILD_ON_DARWIN
+NX_BUILD_ON_SUN
+NX_BUILD_ON_FreeBSD
+
+# Build PIC libraries.
+
+if test "$CYGWIN32" != yes -a "$DARWIN" != yes; then
+  CXXFLAGS="$CXXFLAGS -fPIC"
+  CFLAGS="$CFLAGS -fPIC"
+fi
+
+# Solaris requires the socket and gcc_s libs explicitly linked.
+# Note also that headers from default /usr/openwin/include/X11
+# cause a warning due to pragma in Xmd.h.
+
+if test "$SUN" = yes; then
+  LIBS="$LIBS -L/usr/sfw/lib -lsocket "
+  CXXFLAGS="$CXXFLAGS -I/usr/sfw/include"
+  CFLAGS="$CFLAGS -I/usr/sfw/include"
+fi
+
+# On FreeBSD search libraries and includes under /usr/local.
+
+if test "$FreeBSD" = yes; then
+  LIBS="$LIBS -L/usr/local/lib"
+  CXXFLAGS="$CXXFLAGS -I/usr/local/include"
+  CFLAGS="$CFLAGS -I/usr/local/include"
+fi
+
+# Under Darwin we don't have support for -soname option and
+# we need the -bundle flag. Under Solaris, instead, we need
+# the options -G -h.
+
+if test "$DARWIN" = yes; then
+  LDFLAGS="$LDFLAGS -bundle"
```

Comment:
> I'd leave out the Darwin and Sun sections completely.

Question: Any idea about the FreeBSD section, then? Why do we want to include stuff from /usr/local? Only reason, I can imagine, that nxcomp builds against something not available on FreeBSD. But do we have a guess on what that could be?

I removed the SUN and DARWIN section now. Waiting for feedback on the FreeBSD section.